src/compiler/preprocess/parse_tag_attributes.ts

```diff
@@ -0,0 +1,14 @@
+export default function parse_tag_attributes(str: string) {
```

Comment:
do we need to put this in its own file since it's such a small function and only used by one other file?

Reply:
I guess there are different ways to think about this. My thinking was that it feels like a "utility" function in that it makes sense in isolation and you probably don't want to see the implementation when you're looking at preprocessing code (which still a rather long file). I personally don't see any downside to have a lot of small files.

Reply:
ah, I was thinking that when it's in its own file it gets `export`ed, so it's harder to tell where it's used vs if it's in the same fine then it's not `export`ed so clear it's only used in that file

Reply:
ok, it's true that since there is no package scope in JS, the overall API surface gets bigger and I guess that could also be an issue when minifying/tree shaking. ok, i'll inline it again.